libnetwork/controller.go

```diff
@@ -392,14 +392,14 @@ func (c *Controller) processNodeDiscovery(nodes []net.IP, add bool) {
 	})
 }
 
-func (c *Controller) pushNodeDiscovery(d discoverapi.Discover, cap driverapi.Capability, nodes []net.IP, add bool) {
+func (c *Controller) pushNodeDiscovery(d discoverapi.Discover, capability driverapi.Capability, nodes []net.IP, add bool) {
 	var self net.IP
 	// try swarm-mode config
-	if agent := c.getAgent(); agent != nil {
```

Comment:
Perhaps I should do the reverse, and rename the `agent` type. I see it's used in many places, so perhaps better to rename the type.